backend/src/main/java/dccs/academy/entities/DepartmentEntity.java

```diff
@@ -0,0 +1,58 @@
+package dccs.academy.entities;
+
+import jakarta.persistence.*;
+
+import java.time.LocalDateTime;
+
+
+@Entity
+@Table(name = "departments",schema = "certificates")
+public class DepartmentEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id",unique = true,nullable = false)
+    private Long id;
+
+    @Column(name = "title", nullable = false, unique = true)
+    private String title;
+
+    @Column(name = "created_at",updatable = false)
+    private LocalDateTime createdAt;
+
+    @Column(name = "updated_at")
+    private LocalDateTime updatedAt;
+
+    @PrePersist
+    private void onCreate(){
```

Comment:
You have this repeating in every entity. You could implement BaseEntity and store pre-persist and pre-update logic to reduce code duplications. Others entities would extend base one.